rewrite-java/src/main/java/org/openrewrite/java/ChangePackage.java

```diff
@@ -349,4 +383,24 @@ private boolean isTargetRecursivePackageName(String packageName) {
         }
 
     }
+
+    @Value
+    @EqualsAndHashCode(callSuper = false)
+    private static class ReferenceChangePackageVisitor extends TreeVisitor<Tree, ExecutionContext> {
+        Set<Reference> matches;
```

Comment:
I would instead declare a field like this:
```suggestion
        Map<Tree, Reference> matches;
```
The way that `Tree` is required to override `equals()` and `hashCode()` you can then call `get()` on the map.